features/FxA-89-devices-view/README.md

```diff
@@ -0,0 +1,406 @@
+
+We will implement this feature in several phases,
+intending each phase to be independently shippable
+if we so choose.
+
+Since the primary goal for this feature is security,
+we will first focus on implementing the parts
+related to disconnecting devices.
+This will be followed by additional notifications
+and security checks to improve the disconnection experience,
+and finally by the new affordances
+for connecting additional devices.
+
+### Phase 1
+
+First we'll make sure
+that the core disconnecting-your-devices experience
+works well and emits the necessary metrics,
+building on the existing devices view code.
+
+Acceptance criteria:
+
+* [ ] When I am connected to Sync on Desktop Firefox,
+      my device list shows that device
+      with a desktop icon,
+      a plausible last-active timestamp,
+      and a "Disconnect..." button.
+* [ ] When I am connected to Sync from Fennec or iOS on a phone,
+      my device list shows that device
+      with a phone icon,
+      a plausible last-active timestamp,
+      and a "Disconnect..." button.
+* [ ] When I am connected to Sync from Fennec or iOS on a tablet,
+      my device list shows that device
+      with a tablet icon,
+      a plausible last-active timestamp,
+      and a "Disconnect..." button.
+* [ ] When I sign in to Firefox Accounts on the web
+      without connecting to sync,
+      this does not appear as an item
+      in the device list.
+* [ ] The devices view does not show
+      any messaging about connecting new devices.
+* [ ] Clicking the "Disconnect..." button for a device
+      causes a modal confirmation dialog to be displayed
+      with the specified messaging.
+* [ ] Successfully submitting the dialog
+      destroys the device record and session
+      token for that device.
+* [ ] Front-end metrics events are emitted
+      when users view the devices list,
+      click the "Disconnect..." button,
+      submit or cancel the disconnect dialog,
+      click the "refresh" button,
+      and click the "missing or duplicate devices" link.
+* [ ] Back-end metrics events are emitted
+      when a device is disconnected,
+      and include the reason for its disconnection.
+* [ ] Graphs are available to view
+      the number of views of this screen,
+      the number of device disconnections
+      broken down by reason,
+      and the click rate on various buttons and links.
+
+With these parts of the feature built,
+we should consider enabling it by default
+for some percentage of the userbase,
+in order to get preliminary metrics
+on whether and how it will be used.
+
+### Phase 2
+
+Next we add some of the trickier parts
+of the feature,
+that enhance the experience
+but are not expected to make a big difference
+to the core success metrics.
+
+Acceptance criteria:
+
+* [ ] When I click the "Disconnect..." button
+      and my login session is more than 30 minutes old,
+      I required to re-enter my password
+      before I can disconnect the device.
```

Comment:
> I'm guessing password change or account deletion are much more malicious actions than device disconnect.

On reflection, both of those actions require you to provide the password directly as part of the API request, so requiring the user to also refresh their session could be more annoying than helpful.

Given that:

1. The impact of someone maliciously disconnecting one of your devices is pretty small - you just have to log back in again.  Annoying, but not a disaster.
2. It would indeed to cleaner to apply this logic to the whole settings page rather than just the device-management code.

I'm leaning towards pulling this out into its own generalized feature, and shipping the rest of this view without it.  @davismtl what do you think?